<?xml version="1.0"?>
<launch>

	<!-- dirt detection software -->
	<node pkg="ipa_dirt_detection" ns="dirt_detection_client_preprocessing" type="dirt_detection_client_preprocessing" name="dirt_detection_client_preprocessing" output="screen">	<!-- ns=namespace (arbitrary), type=name of executable, name=node name (arbitrary) -->
		<rosparam command="load" file="$(find ipa_dirt_detection)/ros/launch/dirt_detection_client_preprocessing_params.yaml"/>
		
		<!--launch-prefix="/usr/bin/gdb"-->
		<!--remap from="image_color" to="/camera/rgb/image"/-->
		
		<remap from="colored_point_cloud" to="/camera/depth_registered/points"/> <!-- depth_registered -->
		<!--remap from="colored_point_cloud" to="/camera1_pcl"/--> <!-- depth_registered -->
		<remap from="image_color" to="/camera/rgb/image_raw"/> 
		
		<!--remap from="colored_point_cloud" to="/cam3d/depth_registered/points"/-->
		<!--remap from="colored_point_cloud" to="/cam3d/rgb/points"/-->
		<!--remap from="colored_point_cloud" to="/camera/rgb/points"/-->
		<!--remap from="colored_point_cloud" to="/cam3d/rgb/points"/-->
		
		<!-- if not an empty string, this enables using a mask that defines areas in the map where dirt detections are valid (i.e. this mask can be used to exclude areas from dirt mapping, white=detection area, black=do not detect) -->
		<!-- param name="dirtMappingMaskFilename" value="$(find ipa_dirt_detection)/common/files/map_masks/" /-->
	</node>

	<node pkg="ipa_dirt_detection" ns="dirt_detection_server_spectral" type="dirt_detection_server_spectral" name="dirt_detection_server_spectral" output="screen" >
		<rosparam command="load" file="$(find ipa_dirt_detection)/ros/launch/dirt_detection_server_spectral_params.yaml"/>
	</node>

</launch>
